<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>StackerWidget</class>
 <widget class="QWidget" name="StackerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>350</width>
    <height>564</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>350</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <property name="margin">
    <number>3</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Stacker</string>
     </property>
     <property name="flat">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="margin">
       <number>6</number>
      </property>
      <property name="spacing">
       <number>6</number>
      </property>
      <item row="30" column="0" colspan="7">
       <widget class="QGroupBox" name="previewBox">
        <property name="title">
         <string>Stacking Preview</string>
        </property>
        <property name="flat">
         <bool>true</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Resolution</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="hidderViewerSize">
           <property name="suffix">
            <string> px</string>
           </property>
           <property name="minimum">
            <number>10</number>
           </property>
           <property name="maximum">
            <number>1024</number>
           </property>
           <property name="value">
            <number>400</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>#Object</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3">
          <widget class="QSpinBox" name="stackCount">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="value">
            <number>3</number>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="26" column="0" colspan="7">
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Stacking configuration</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_5">
         <item row="2" column="1" colspan="4">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>Best stacking direction</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1" colspan="3">
          <widget class="QCheckBox" name="rotAroundAxis">
           <property name="text">
            <string>Rot around  line</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Axis</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QLabel" name="label_12">
           <property name="text">
            <string>Density</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QSpinBox" name="searchDensity">
           <property name="minimum">
            <number>2</number>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item row="5" column="2">
          <widget class="QSpinBox" name="searchAxis">
           <property name="maximum">
            <number>2</number>
           </property>
          </widget>
         </item>
         <item row="5" column="3" colspan="2">
          <widget class="QCheckBox" name="searchAlongAxis">
           <property name="text">
            <string>along axis</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="5" column="6">
          <widget class="QPushButton" name="searchDirectionButton">
           <property name="text">
            <string>Search..</string>
           </property>
          </widget>
         </item>
         <item row="3" column="3" colspan="2">
          <widget class="QLabel" name="label_13">
           <property name="text">
            <string>Angle density</string>
           </property>
          </widget>
         </item>
         <item row="3" column="6">
          <widget class="QSpinBox" name="searchAngleDensity">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>360</number>
           </property>
           <property name="value">
            <number>8</number>
           </property>
          </widget>
         </item>
         <item row="7" column="1" colspan="6">
          <widget class="Line" name="line_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="8" column="4">
          <widget class="QCheckBox" name="rotFreeForm">
           <property name="text">
            <string>Free Rot</string>
           </property>
          </widget>
         </item>
         <item row="8" column="6">
          <widget class="QSpinBox" name="rotStackingDensity">
           <property name="value">
            <number>12</number>
           </property>
          </widget>
         </item>
         <item row="8" column="5">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Density</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="24" column="0" colspan="7">
       <widget class="QGroupBox" name="controllersPanel">
        <property name="title">
         <string>Controllers</string>
        </property>
        <property name="flat">
         <bool>false</bool>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="3" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>Axis</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="convertGcAxis">
           <property name="maximum">
            <number>2</number>
           </property>
           <property name="value">
            <number>2</number>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>#Joint</string>
           </property>
          </widget>
         </item>
         <item row="3" column="3">
          <widget class="QSpinBox" name="skeletonJoints">
           <property name="singleStep">
            <number>2</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item row="3" column="4" colspan="2">
          <widget class="QCheckBox" name="basicFitGC">
           <property name="text">
            <string>Basic</string>
           </property>
          </widget>
         </item>
         <item row="3" column="6" colspan="2">
          <widget class="QPushButton" name="convertToGC">
           <property name="text">
            <string>GC</string>
           </property>
           <property name="icon">
            <iconset resource="../Workspace.qrc">
             <normaloff>:/Workspace/Resources/icons/gc.png</normaloff>:/Workspace/Resources/icons/gc.png</iconset>
           </property>
          </widget>
         </item>
         <item row="10" column="2" colspan="2">
          <widget class="QCheckBox" name="useAABB">
           <property name="text">
            <string>Use AABB</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="10" column="0" colspan="2">
          <widget class="QComboBox" name="cuboidMethod">
           <item>
            <property name="text">
             <string>MinOBB3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>OBB</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>OBB2</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="10" column="6" colspan="2">
          <widget class="QPushButton" name="convertToCuboid">
           <property name="text">
            <string>Cuboid</string>
           </property>
           <property name="icon">
            <iconset resource="../Workspace.qrc">
             <normaloff>:/Workspace/Resources/icons/cuboid.png</normaloff>:/Workspace/Resources/icons/cuboid.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="25" column="0" colspan="7">
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Joint</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_4">
         <item row="0" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Voxel size</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QPushButton" name="findPairwiseJointsButton">
           <property name="text">
            <string>Pairwise Joints</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="findJointsButton">
           <property name="text">
            <string>Find Joints</string>
           </property>
           <property name="icon">
            <iconset resource="../Workspace.qrc">
             <normaloff>:/Workspace/Resources/icons/joints.png</normaloff>:/Workspace/Resources/icons/joints.png</iconset>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="numJoints"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string># Joints</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="jointsThreshold">
           <property name="decimals">
            <number>5</number>
           </property>
           <property name="minimum">
            <double>0.000010000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>0.035000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="13" column="4">
       <widget class="QPushButton" name="saveSuggestionsButton">
        <property name="text">
         <string>Save</string>
        </property>
        <property name="autoRepeatDelay">
         <number>300</number>
        </property>
       </widget>
      </item>
      <item row="13" column="2">
       <widget class="QLabel" name="numSuggestion">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="11" column="5">
       <widget class="QLabel" name="numSolution">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QSpinBox" name="suggestionID">
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item row="11" column="3">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Solution:</string>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Suggestion:</string>
        </property>
       </widget>
      </item>
      <item row="11" column="4">
       <widget class="QSpinBox" name="solutionID">
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="13" column="5">
       <widget class="QPushButton" name="loadSuggestionsButton">
        <property name="text">
         <string>Load</string>
        </property>
       </widget>
      </item>
      <item row="13" column="3">
       <widget class="QPushButton" name="suggestButton">
        <property name="text">
         <string>Suggest</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <widget class="QPushButton" name="improveButton">
        <property name="text">
         <string>Improve</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="targetStackLabel">
        <property name="text">
         <string>TargetS</string>
        </property>
       </widget>
      </item>
      <item row="7" column="3">
       <widget class="QDoubleSpinBox" name="BBTolerance">
        <property name="minimum">
         <double>0.200000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QDoubleSpinBox" name="targetS">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>0.700000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>BB toleren</string>
        </property>
       </widget>
      </item>
      <item row="7" column="4">
       <widget class="QLabel" name="label_17">
        <property name="text">
         <string># Expt sln</string>
        </property>
       </widget>
      </item>
      <item row="7" column="5">
       <widget class="QSpinBox" name="numExpectedSolutions">
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Output for paper</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2" colspan="2">
       <widget class="QLineEdit" name="outputFileName">
        <property name="text">
         <string>result_0</string>
        </property>
       </widget>
      </item>
      <item row="0" column="4" colspan="2">
       <widget class="QPushButton" name="outputButton">
        <property name="text">
         <string>Output</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="6">
       <widget class="Line" name="line">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="2" column="4" colspan="2">
       <widget class="QPushButton" name="hotspotsButton">
        <property name="text">
         <string>Hot Spots</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../Workspace.qrc"/>
 </resources>
 <connections/>
</ui>
